### ISSUE TITLE
Refactor kafka config

### ISSUE THREAD

**Issue by mkmik** [influxdata/influxdb_iox#1946]
TODO: reword this into english

Currently DatabaseRules have a `routing_config` field:

```
    // Routing config
    RoutingConfig routing_config = 9;
```

which is currently:

```
message RoutingConfig {
  NodeGroup target = 1;
}
```

Let's turn it into a:

```
message RoutingConfig {
  oneof sink {
    NodeGroup target = 1;
    KafkaTarget kafka = 2;
  }
}

message KafkaTarget {
  // if empty defaults to the database name
  string topic = 1;
}
```

Similarly add a "subscription" config to tell a query node to read from kafka.

Then, move the "kafka connection string" as part of the node config (like the objectstore config), not the (persisted) database rules.

---

This would replace the current :

```
  // Optionally, the address of the write buffer for writing or reading/restoring data
  oneof write_buffer_connection {
    string writing = 11;
    string reading = 12;
  }
```

CC: @pauldix

**Comment by carols10cents** [comment]
I was trying to keep the word "Kafka" out as much as possible to make room for a different write buffer implementation in the future; this might be the place it has to say "Kafka" though.

**Comment by crepererum** [comment]
Fixed by #2458.
